youtube_dl/extractor/voe.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    js_to_json,
+    url_or_none,
+)
+
+
+class VOEIE(InfoExtractor):
+    IE_NAME = 'voe'
+    IE_DESC = 'VOE.SX'
+    _VALID_URL = r'https?://voe\.sx/(e/)?(?P<id>[a-z0-9]+)'
+    _TEST = {
+        'url': 'https://voe.sx/e/ng7ja5n5n2y8',
+        'info_dict': {
+            'id': 'ng7ja5n5n2y8',
+            'title': 'md5:a86687fb962742f04652aee19ad34e06',
+            'thumbnail': r're:^https?://.*\.jpg$',
+            'ext': 'm3u8',
+        },
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(
+            'https://voe.sx/e/%s' % video_id, video_id)
+
+        sources = self._parse_json(
+            self._search_regex(r'\bsources\s*=\s*(\{[^}]+\})', webpage, 'sources'),
+            video_id, transform_source=js_to_json)
+
+        title = self._search_regex(
+            r'<title>(?:Watch\s+)?(?P<title>.+?)(?:-\s+VOE\s+\|.+)?</title>',
+            webpage, 'title', group='title')
+
+        formats = []
+
+        f_url = url_or_none(sources.get('hls'))
```

Comment:
The video links may be base64-encoded:
```suggestion
        for fmt in ('mp4, 'hls'):
            if fmt not in sources:
                continue
            sources[fmt] = url_or_none(sources['fmt']) or url_or_none(base64.b64decode(sources['fmt']))
            f_url = sources.get('hls')
```